bindings/go/dag/dag.go

```diff
@@ -89,12 +113,9 @@ func (d *DirectedAcyclicGraph[T]) DeleteVertex(id T) error {
 	}
 
 	// Remove all edges to this node
-	for _, node := range d.Vertices {
-		if _, exists := node.Edges[id]; exists {
-			// Decrement the in-degree of the node
-			d.Vertices[node.ID].InDegree--
-			// Remove the edge from the node
-			delete(node.Edges, id)
+	for edge := range d.Vertices[id].Edges {
```

Comment:
Do we need to consider updating edges that point to the deleted id?

We don't, right? Because we don't want to destroy the dag, right?

Reply:
yeah the idea here is to not modify the input graph anymore